internal/pkg/test/tool/require/require_linux.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package require
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/pkg/network"
+)
+
+var hasUserNamespace bool
+var hasUserNamespaceOnce sync.Once
+
+var supportNetwork bool
+var supportNetworkOnce sync.Once
+
+// UserNamespace checks that the current test could use
+// user namespace, if user namespaces are not enabled or
+// supported, the current test is skipped with a message.
+func UserNamespace(t *testing.T) {
+	// not performance critical, just save extra execution
+	// to get the same result
+	hasUserNamespaceOnce.Do(func() {
+		// user namespace is a bit special, as there is no simple
+		// way to detect if it's supported or enabled via a call
+		// on /proc/self/ns/user, the easiest and reliable way seems
+		// to directly execute a command by requesting user namespace
+		cmd := exec.Command("/bin/true")
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Cloneflags: syscall.CLONE_NEWUSER,
+		}
+		// no error means user namespaces are enabled
+		hasUserNamespace = cmd.Run() == nil
+	})
+	if !hasUserNamespace {
+		t.Skipf("user namespaces seems not enabled or supported")
+	}
+}
+
+// Network check that bridge network is supported by
```

Comment:
Why do we require that the bridge network is enabled? Just testing for CNI?

I figured this would just test for the network namespace working.

Could this be split into multiple functions? 1 for the namespace require, 1 for the network require?

Reply:
What the reason of splitting it into two functions ? If we check for network namespace it means we are going to use `network` package with CNI

Reply:
I guess I thought checking if the network namespace is an enabled feature should be a smaller function we could maybe reuse, but you're correct in that if it is enabled, we would use CNI anyways.